docs/smart-contracts/guides/conventions/work-contractspec-js.mdx

```diff
@@ -4,4 +4,696 @@ hide_table_of_contents: true
+
+Now you have the complete contract specification in JavaScript. You can use this specification to interact with your Soroban smart contract.
+
+With these steps, you have learned how to define a contract specification from scratch using JavaScript. You can proceed with deploying the contract, generating TypeScript bindings, and using the specification to interact with the contract.
+
+#### Option 2 Get the Spec from a Deployed Contract.
+
+##### Step 1: Deploy the Contract
+
+First, deploy your Soroban smart contract to the network.
+
+```bash
+soroban contract deploy --source orbit --network testnet --wasm token_manager.wasm
+```
+
+This command will return the contract ID, for example: `CBEUVPRZKMCX5MPUAF2JMKMRIFEXXJXQXBLUUW6LMN5GDDQQSX3UHYXA`.
+
+##### Step 2: Generate TypeScript Bindings
+
+Use the Soroban CLI to generate TypeScript bindings which use the contract's schema specifications to provide type-checked contract methods.
+
+```bash
+soroban contract bindings typescript --output-dir testbindings --network testnet --contract-id CBEUVPRZKMCX5MPUAF2JMKMRIFEXXJXQXBLUUW6LMN5GDDQQSX3UHYXA
+```
+
+This command will create TypeScript bindings in the specified directory.
+
+Alternatively, you can get your spec using javascript: credit to [@mason in Stellar Developers discord](https://discord.com/channels/897514728459468821/966788672164855829/1201708320336138271) and [his gist here] (https://gist.github.com/masonforest/27ebfafa3452227937b842c56731d817)
+
+Save this file as `readContractSpec.js`
+
+```typescript
+import fs from "node:fs/promises";
+import { Contract, xdr } from "@stellar/stellar-sdk";
+import jsXdr from "@stellar/js-xdr";
+
+const { XdrReader } = jsXdr;
+
+async function readContractSpec(path) {
+  const wasmBuffer = await fs.readFile(path);
+  const wasmModule = await WebAssembly.compile(wasmBuffer);
+  const reader = new XdrReader(
+    Buffer.from(
+      WebAssembly.Module.customSections(wasmModule, "contractspecv0")[0],
+    ),
+  );
+  const specs = [];
+
+  do {
+    specs.push(xdr.ScSpecEntry.read(reader));
+  } while (!reader.eof);
+
+  return new Contract.Spec(specs);
+}
+
+async function main() {
+  const wasmFilePath = process.argv[2];
+
+  if (!wasmFilePath) {
+    console.error("Usage: node readContractSpec.js <wasmFilePath>");
+    process.exit(1);
+  }
+
+  try {
+    const contractSpec = await readContractSpec(wasmFilePath);
+    console.log("Contract Specification:", contractSpec);
+  } catch (error) {
+    console.error("Error reading contract specification:", error);
+  }
+}
+
+main();
+```
+
+### Running the Script
+
+To run the script with `mycontract.wasm`, use the following command in your terminal:
+
+```bash
+node readContractSpec.js mycontract.wasm
+```
+
+This will execute the `readContractSpec` function with the `mycontract.wasm` file provided as a command-line argument and output the contract specification to the console. Copy the output for the next step
```

Comment:
Instead of making it a standalone CLI tool, you could just incorporate it into the next step as a code-only step.